app/assets/javascripts/admin/components/admin-web-hook-event-chooser.js.es6

```diff
@@ -0,0 +1,44 @@
+import computed from 'ember-addons/ember-computed-decorators';
+
+export default Ember.Component.extend({
+  tagName: '',
+  typeName: Ember.computed.alias('type.name'),
+
+  @computed('typeName')
+  name(typeName) {
+    return I18n.t(`admin.web_hooks.${typeName}_event.name`);
+  },
+
+  @computed('typeName')
+  details(typeName) {
+    return I18n.t(`admin.web_hooks.${typeName}_event.details`);
+  },
+
+  @computed('model.[]')
+  eventTypeExists(eventTypes) {
+    return eventTypes.any(event => event.name === this.get('typeName'));
+  },
+
+  @computed
+  enabled: {
+    get() {
+      return this.get('eventTypeExists');
+    },
+    set(value) {
+      const type = this.get('type');
+      const model = this.get('model');
+      // add an association when not exists
+      if (value !== this.get('eventTypeExists')) {
+        if (value) {
+          model.addObject(type);
+        } else {
+          model.removeObjects(model.filter(eventType => {
+            return (eventType.name || eventType.get('name')) === (type.name || type.get('name'));
```

Comment:
They should always be an Ember object. `model`(array) and `type`(an event) are both Ember objects. After `filter`, `eventType` is not an Ember object any more. I can simply use the simple `.` here since there isn't an computed property.